scripts/check.sh

```diff
@@ -187,9 +191,24 @@ if [[ "${START_REVISION}" == *..* ]]; then
     git fetch origin
   fi
 
-  NEW_RANGE_START=$(git merge-base origin/master "${RANGE_END}")
-  START_REVISION="${START_REVISION/$RANGE_START/$NEW_RANGE_START}"
-  START_SHA="${START_REVISION}"
+  # Try to come up with a more accurate representation of the merge, so that
+  # checks will operate on just the differences the PR would merge into master.
+  # The start of the revision range that Travis supplies can sometimes be a
+  # seemingly random value.
+  NEW_RANGE_START=$(git merge-base origin/master "${RANGE_END}" || echo "")
+  if [[ -n "$NEW_RANGE_START" ]]; then
+    START_REVISION="${NEW_RANGE_START}..${RANGE_END}"
```

Comment:
Why is the then code different from the old version?

Reply:
This more directly expresses the intent: previously this was substituting the new start into the old range string. This is now just constructing a new range from the new start and the old end.